# HG changeset patch
# User Sam Lantinga <slouken@libsdl.org>
# Date 1592588234 25200
#      Fri Jun 19 10:37:14 2020 -0700
# Node ID fb6395c49ce97ad3f70de2dfd616c715ab0ebc6d
# Parent  b622185b81ce31ac8e0c418f7d2cefd3feb6ffd4
Fixed bug 5199 - Fix KMSDRM_CreateWindow() segfault when starting LÖVE2D engine.

Manuel Alfayate Corchete

This small patch fixes the KMSDRM_CreateSurfaces() call in KMSDRM_CreateWindow(), that was segfaulting deeper into SDL internals because the windata->viddata pointer wasn't set before the KMSDRM_CreateSurfaces() call.
So that's what this small patch does.

Now, LÖVE2D works perfectly well on the Raspberry Pi 3, instead of just segfaulting.

diff --git a/src/video/kmsdrm/SDL_kmsdrmvideo.c b/src/video/kmsdrm/SDL_kmsdrmvideo.c
--- a/src/video/kmsdrm/SDL_kmsdrmvideo.c
+++ b/src/video/kmsdrm/SDL_kmsdrmvideo.c
@@ -798,6 +798,7 @@ KMSDRM_CreateWindow(_THIS, SDL_Window * 
     }
 
     /* Setup driver data for this window */
+    windata->viddata = viddata;
     window->driverdata = windata;
 
     if (KMSDRM_CreateSurfaces(_this, window)) {
@@ -807,8 +808,6 @@ KMSDRM_CreateWindow(_THIS, SDL_Window * 
     /* Add window to the internal list of tracked windows. Note, while it may
        seem odd to support multiple fullscreen windows, some apps create an
        extra window as a dummy surface when working with multiple contexts */
-    windata->viddata = viddata;
-
     if (viddata->num_windows >= viddata->max_windows) {
         int new_max_windows = viddata->max_windows + 1;
         viddata->windows = (SDL_Window **)SDL_realloc(viddata->windows,
